compiler/packages/babel-plugin-react-compiler/src/HIR/CollectHoistablePropertyLoads.ts

```diff
@@ -194,7 +194,10 @@ type PropertyPathNode =
 class PropertyPathRegistry {
   roots: Map<IdentifierId, RootNode> = new Map();
 
-  getOrCreateIdentifier(identifier: Identifier): PropertyPathNode {
+  getOrCreateIdentifier(
+    identifier: Identifier,
+    reactive: boolean,
+  ): PropertyPathNode {
```

Comment:
Note that we now need to track reactivity, as we're creating new instructions to represent dependencies